opm/autodiff/SimulatorFullyImplicitBlackoilOutput.cpp

```diff
@@ -235,52 +235,75 @@ namespace Opm
     void
     BlackoilOutputWriter::
     writeTimeStep(const SimulatorTimerInterface& timer,
-                  const SimulatorState& state,
-                  const WellState& wellState,
+                  const SimulatorState& localState,
+                  const WellState& localWellState,
                   bool substep)
     {
-        // VTK output
+        // VTK output (is parallel if grid is parallel)
         if( vtkWriter_ ) {
-            vtkWriter_->writeTimeStep( timer, state, wellState , false );
+            vtkWriter_->writeTimeStep( timer, localState, localWellState, false );
         }
-        // Matlab output
-        if( matlabWriter_ ) {
-            matlabWriter_->writeTimeStep( timer, state, wellState , false );
-        }
-        // ECL output
-        if ( eclWriter_ ) {
-            eclWriter_->writeTimeStep(timer, state, wellState, substep);
-        }
-        // write backup file
-        if( backupfile_ )
+
+        if( parallelOutput_ )
```

Comment:
This confused me a little: it seems to be true (non-null) exactly when output_ also is true, so in the serial case this is true, too. I think it would be better to use if (output_) here.